pkg/plugin/plugin.go

```diff
@@ -118,15 +122,50 @@ func makeAutoscaleEnforcerPlugin(
 		}
 	}
 
+	// A note about event handling:
+	//
+	// Before returning from this function, we want to make sure that we're caught up to the watch
+	// events generated by initially reading the cluster state (i.e. the initial List()).
+	//
+	// Doing this is non-trivial, so we accomplish it in pieces:
+	//
+	// 1. Using watch.WatchModeSync to force queueing events *before* returning from creating the
+	//    watcher (note: and therefore, before any start to be handled); and
+	// 2. For each event created from the initial List(), increment a counter to track the number of
+	//    these events, and decrement it as events are handled.
+	//
+	// The initial state building is complete when the counter reaches zero, at which point we close
+	// the channel that this function will wait on.
+
+	var initEventsCount atomic.Int32
+	initEventsDone := make(chan struct{})
+	incEventCountIfInit := func(isInit bool) {
+		if isInit {
+			initEventsCount.Add(1)
+		}
+	}
+	decEventCountIfInit := func(isInit bool) {
+		if isInit {
+			newCount := initEventsCount.Add(-1)
+			if newCount == 0 {
+				close(initEventsDone)
+			}
+		}
+	}
+
 	hlogger := logger.Named("handlers")
 	nwc := nodeWatchCallbacks{
 		submitNodeDeletion: func(logger *zap.Logger, nodeName string) {
 			pushToQueue(logger, nodeName, func() { p.handleNodeDeletion(hlogger, nodeName) })
 		},
 	}
 	pwc := podWatchCallbacks{
-		submitStarted: func(logger *zap.Logger, pod *corev1.Pod) {
-			pushToQueue(logger, pod.Name, func() { p.handleStarted(hlogger, pod) })
+		submitStarted: func(logger *zap.Logger, pod *corev1.Pod, preexisting bool) {
+			incEventCountIfInit(preexisting)
```

Comment:
Why passing bool instead of an if here?

Reply:
Initially it was because I thought there'd be more than one callback that would be involved here, and that would keep callback functions small.

It's now clear that's not necessary though, so moved the if here in 564ec88212cc6571e3620737a1d48450850b2174